bokeh/palettes.py

```diff
@@ -486,14 +510,21 @@
     "Viridis" : { 3: Viridis3, 4: Viridis4, 5: Viridis5, 6: Viridis6, 7: Viridis7, 8: Viridis8, 9: Viridis9,  10: Viridis10, 11: Viridis11 },
 })
 
-def _cmap_func_generator(name, cmap):
+def _linear_cmap_func_generator(name, cmap):
     def func(n):
-        """Return a palette of n equally spaced colors."""
+        if n>len(cmap): raise ValueError("Requested %(r)s colors, function can only return colors up to the base palette's length (%(l)s)" % 
+            {'r':n,'l':len(cmap)})
         return [cmap[int(math.floor(i))] for i in np.linspace(0, len(cmap)-1, num=n)]
     func.__name__ = name
     return func
 
-_cmaps = {name: _cmap_func_generator(name, cmap) for name, cmap in (('magma', Magma256),
+def grey_func(n):
```

Comment:
I'm sure I missed the discussion - but what's the logic for breaking this into a different function from `_linear_cmap_func_generator`?